lib/ide/src/transport/web.rs

```diff
@@ -0,0 +1,250 @@
+//! web_sys::WebSocket-based `Transport` implementation.
+
+use crate::prelude::*;
+
+use basegl_system_web::closure::ClosureStorage;
+use failure::Error;
+use futures::channel::mpsc;
+use json_rpc::Transport;
+use json_rpc::TransportEvent;
+use web_sys::CloseEvent;
+use web_sys::Event;
+use web_sys::MessageEvent;
+
+
+
+// =================
+// === Utilities ===
+// =================
+
+#[wasm_bindgen]
+extern "C" {
+    /// Converts given `JsValue` into a `String`.
+    #[allow(unsafe_code)]
+    #[wasm_bindgen(js_name="String")]
+    fn js_to_string(s: JsValue) -> String;
```

Comment:
Does `dyn_into` or `unsafe_dyn_into` not allow for that? I like the name `unsafe_dyn_into` more as this is very unsafe. If you provide here nything which is not a string it will blow up.

Reply:
Why is this unsafe? https://www.w3schools.com/jsref/jsref_string.asp 
This show quite numerous examples of calling with not a strings.

I believe `dyn_into`  is about low level casting, also no instance for String. (only JsString that then would need to be processed further to deal with unpaired utf16 surrogates)

Reply:
Oh, I totally missed this part `js_name="String"`. All good! In such a case, this should be refacored to lib/system/web